Tasks/VsTest/vstest.ts

```diff
@@ -619,6 +654,13 @@ function invokeVSTest(testResultsDirectory: string): Q.Promise<number> {
         .then(function (overriddenSettingsFile) {
             locateVSVersion()
                 .then(function (vsVersion) {
+                    try {
+                        vstestLocation = getVSTestLocation(vsVersion);
+                    } catch (e) {
+                        tl.error(e.message);
+                        defer.resolve(1);
```

Comment:
why to create defer. directly return Promise value if you have nothing to do after resolving